### PR TITLE
improve ModalSelectorLayout

### PR DESCRIPTION
**Pull request by brumeregan**
Hi there!
Please review this PR that improves functionality of <ModalSelectorLayout />

**belowNode** and **showBelowNodeOnSelect** were added to achieve this design https://app.goabstract.com/projects/05528230-bb19-11e8-9f9f-c92dc976b27b/branches/master/collections/fbd9fe25-cee1-4f7f-861e-ba5f6d97e0f0?collectionLayerId=9b4fbfbf-bdfe-41a7-85bd-258001ab663d&mode=design

**disableConfirmation** - to disable Ok button as on common modals
**onSelect** - callback that returns selected item. (mostly like onChange for Radio.Group)
**subtitleNode** - for custom subtitleNode, when it is not simple text

Also, I added **styles for scrollbar**

I changed layout styles for <Selector /> component, because flex styles not allowed to insert below component. So not to break existed styles, I created additional wrapper

### ✅ Checklist

- 📚 Change is documented
  - [ ] Other (explain)

### CODE REVIEW

**Comment by mykas** [comment]
I am not able to see the design from the link, can you share a screenshot?

**Comment by brumeregan** [comment]
Screen Shot 2019-11-04 at 11 13 33 AM

File: src/ModalSelectorLayout/ModalSelectorLayout.js [hunk 1]
```diff
@@ -186,7 +193,11 @@ export default class ModalSelectorLayout extends WixComponent {
     } = this.state;
 
     return (
-      <div className={css.modalContent} style={{ height, maxHeight }}>
+      <div
+        className={css.modalContent}
+        style={{ height, maxHeight }}
+        data-hook="modal-content"
```
Comment (mykas) [inline src/ModalSelectorLayout/ModalSelectorLayout.js]: Why this datahook has been removed?

File: src/ModalSelectorLayout/ModalSelectorLayout.js [hunk 2]
```diff
@@ -288,15 +299,22 @@ export default class ModalSelectorLayout extends WixComponent {
...
+              subtitleNode={item.subtitleNode}
+              belowNode={item.belowNode}
+              showBelowNodeOnSelect={item.showBelowNodeOnSelect}
               isSelected={isSelected(item)}
               isDisabled={item.disabled}
-              onToggle={() => !item.disabled && onToggle(item)}
+              onToggle={() => {
+                !item.disabled && onToggle(item);
+
+                if (onSelect) {
```
Comment (mykas) [inline src/ModalSelectorLayout/ModalSelectorLayout.js]: This should be added to `onToggle` function. Please extract this and do it in one place.

File: src/ModalSelectorLayout/ModalSelectorLayout.scss [hunk 3]
```diff
@@ -13,6 +14,8 @@
   flex: 1;
   overflow: auto;
 
+  @include default-scroll-bar;
```
Comment (mykas) [inline src/ModalSelectorLayout/ModalSelectorLayout.scss]: This change should be extracted to different pr

File: src/Selector/Selector.js [hunk 4]
```diff
@@ -28,6 +28,9 @@ class Selector extends WixComponent {
     extraNode: PropTypes.node,
     onToggle: PropTypes.func,
     toggleType: PropTypes.oneOf(['checkbox', 'radio']),
+    showBelowNodeOnSelect: PropTypes.bool,
```
Comment (mykas) [inline src/Selector/Selector.js]: All these changes must be extracted to different pr because this will also have effect on EditableSelector component.

File: src/ModalSelectorLayout/ModalSelectorLayout.js (hunk 1 above)
Comment (brumeregan) [reply src/ModalSelectorLayout/ModalSelectorLayout.js]: data-hook was added, it wasn't previously.
I decided to add the same data-hook as in Modal component
To make it more consistency =)

File: src/Selector/Selector.js (hunk 4 above)
Comment (brumeregan) [reply src/Selector/Selector.js]: but still PR to ModalSelectorLayout will depend on PR to Selector.
Ok, I'll split to 3 PR these changes )

File: src/ModalSelectorLayout/ModalSelectorLayout.js (hunk 1 above)
Comment (mykas) [reply src/ModalSelectorLayout/ModalSelectorLayout.js]: You are right. My eyes are fooling me

**Comment by mykas** [comment]
Hey @brumeregan , can you please ask your designer to contact our Design Systems UX regarding this design? She can contact directly @milkyfruit

File: src/ModalSelectorLayout/ModalSelectorLayout.js (hunk 2 above)
Comment (brumeregan) [reply src/ModalSelectorLayout/ModalSelectorLayout.js]: moved

File: src/ModalSelectorLayout/ModalSelectorLayout.scss (hunk 3 above)
Comment (brumeregan) [reply src/ModalSelectorLayout/ModalSelectorLayout.scss]: moved to another PR

**Comment by milkyfruit** [comment]
Added a comment in different PR: https://github.com/wix/wix-style-react/pull/4456#issuecomment-551884222

File: src/ModalSelectorLayout/ModalSelectorLayout.js [hunk 5]
```diff
@@ -288,15 +304,18 @@ export default class ModalSelectorLayout extends WixComponent {
...
-                item.extraNode ? (
-                  item.extraNode
-                ) : (
-                  <Text secondary>{item.extraText}</Text>
-                )
+                item.extraNode
+                  ? item.extraNode
+                  : item.extraText && <Text secondary>{item.extraText}</Text>
               }
+              subtitleNode={item.subtitleNode}
```
Comment (argshook) [inline src/ModalSelectorLayout/ModalSelectorLayout.js]: @brumeregan 
please add new properties to `dataSource` prop comment above. Otherwise once build is green it's good to merge

**Comment by argshook** [comment]
there's one perfer degradation error but it's there only because branch got outdated with master.

i'll merge this tomorrow, we're also planning to release.

**Comment by brumeregan** [comment]
@argshook I updated PR:
- add comment with props to items array in dataSource
- updated branch with master
